templates/report/results.html

```diff
@@ -28,33 +28,102 @@
-        <div class="category">
-            <div class="header c{{ name }} toggle" data-toggle="#crates-{{ name }}">
-                {{ name }} ({{ crates|length }})
-            </div>
-
-            <div class="crates hidden" id="crates-{{ name }}">
-                {% for crate in crates %}
-                    <div class="crate">
-                        <a href="{{ crate.url|safe }}" target="_blank" rel="noopener">{{ crate.name }}</a>
-                        {% for run in crate.runs %}
-                            <span class="run">
-                                {% if run %}
-                                    <b class="r{{ run.res }}"></b>
-                                    <a href="{{ run.log|safe }}/log.txt">
-                                        {{ result_names[run.res] }}
-                                    </a>
-                                {% else %}
-                                    <b class="c{{ crate.res }}"></b>
-                                    {{ crate.res }}
-                                {% endif %}
-                            </span>
+        {% for iter in categories %}
+            {% set name = iter.0 %}
+            {% set crates = iter.1 %}
+            <div class="category">
+                {% if crates.Plain %}
+                    <div class="header c{{ name }} toggle" data-toggle="#crt-{{ name }}">
+                        {{ name }} ({{ crates.Plain|length }})
+                    </div>
+                    <div class="crates hidden" id="crt-{{ name }}">
+                        {% for crate in crates.Plain %}
+                            <div class="crate">
+                                <a href="{{ crate.url|safe }}" target="_blank" rel="noopener">{{ crate.name }}</a>
+                                {% for run in crate.runs %}
+                                    <span class="run">
+                                        {% if run %}
+                                            <b class="r{{ run.res }}"></b>
+                                            <a href="{{ run.log|safe }}/log.txt">
+                                                {{ result_names[run.res] }}
+                                            </a>
+                                        {% else %}
+                                            <b class="c{{ crate.res }}"></b>
+                                            {{ crate.res }}
+                                        {% endif %}
+                                    </span>
+                                {% endfor %}
+                            </div>
                         {% endfor %}
                     </div>
-                {% endfor %}
+                {% elif crates.Tree and crates.Tree.count > 0 %}
+                    <div class="header c{{ name }} toggle" data-toggle="#crt-{{ name }}-tr">
+                        {{ name }}: tree ({{ crates.Tree.count }} root crates, {{info[name]}} {{ name }} crates in total) 
```

Comment:
I'd name this "dependencies" in the report instead of tree, since it's not a tree of all the crates, but it only shows dependencies.